java/src/main/java/scenarios/crewcreation/CharacterCreation.java

```diff
@@ -229,10 +212,27 @@ public static void printAbilityScoreFlavorText(int abilityScoreValue, String abi
         System.out.println();
     }
 
-    public static void printAbilityScoresWithNames(Map<String, Integer> scores) {
-        for (Map.Entry<String, Integer> entry : scores.entrySet()) {
-            System.out.println("[" + entry.getKey() + "]: " + entry.getValue());
+    private static int selectScore (List<Integer> availableScores) {
+        // TODO: Depending on how Lanterna is set up, we should potentially extract the following code into a
+        //  utility method for players to select a choice from an ordered list
+        DataPrinter.printAsOrderedList(availableScores);
+
+        Integer selectedScore = null;
+        while (selectedScore == null) {
+            String invalidInputMessage = "Please enter a number from 1 to " + availableScores.size() + ".";
+
+            try {
+                selectedScore = Integer.parseInt(scanner.nextLine());
+            } catch (NumberFormatException e) {
+                System.out.println(invalidInputMessage);
+                continue;
+            }
+            if (selectedScore < 1 || selectedScore > availableScores.size()) {
+                System.out.println(invalidInputMessage);
+                selectedScore = null;
+            }
         }
+        return selectedScore - 1;
```

Comment:
This is good, just add a comment above the return to explain why we're subtracting 1.

Reply:
Nice comment, just add it **above** the return as stated above.